test/integration/spacecraft.cc

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+#include <gtest/gtest.h>
+
+#include <optional>
+
+#include <gz/msgs/actuators.pb.h>
+#include <gz/msgs/twist.pb.h>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Util.hh>
+#include <gz/msgs/Utility.hh>
+#include <gz/transport/Node.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+
+#include "gz/sim/components/AngularVelocity.hh"
+#include "gz/sim/components/Joint.hh"
+#include "gz/sim/components/JointVelocity.hh"
+#include "gz/sim/components/LinearVelocity.hh"
+#include "gz/sim/components/Link.hh"
+#include "gz/sim/components/Name.hh"
+#include "gz/sim/components/Model.hh"
+#include "gz/sim/components/Pose.hh"
+
+#include "gz/sim/Model.hh"
+#include "gz/sim/Server.hh"
+#include "gz/sim/SystemLoader.hh"
+#include "test_config.hh"
+
+#include "../helpers/Relay.hh"
+#include "../helpers/EnvTestFixture.hh"
+
+using namespace gz;
+using namespace sim;
+using namespace std::chrono_literals;
+
+class SpacecraftTest : public InternalFixture<::testing::Test>
+{
+  protected: std::unique_ptr<Server> StartServer(const std::string &_filePath)
+  {
+    ServerConfig serverConfig;
+    const auto sdfFile = std::string(PROJECT_SOURCE_PATH) + _filePath;
+    serverConfig.SetSdfFile(sdfFile);
+
+    auto server = std::make_unique<Server>(serverConfig);
+    EXPECT_FALSE(server->Running());
+    EXPECT_FALSE(*server->Running(0));
+
+    using namespace std::chrono_literals;
+    server->SetUpdatePeriod(1ns);
+    return server;
+  }
+};
+
+/////////////////////////////////////////////////
+// Test that commanded motor speed is applied
+// See https://github.com/gazebosim/gz-sim/issues/1175
+TEST_F(SpacecraftTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(InputTest))
```

Comment:
Do we know if this is necessary?

Reply:
I've kept it as in the ackermann tests, since these are similar. I can try to remove it to check if it fails on Windows.

Reply:
@arjo129 test fails on Windows - seems like the vehicle did not move at all there. Thoughts?